src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,123 @@
+        registeredUser.setLogin(VALID_LOGIN);
+        registeredUser.setPassword(VALID_PASSWORD);
+        registeredUser.setAge(VALID_AGE);
+        storageDao.add(registeredUser);
+        User newUser = new User();
+        newUser.setLogin(VALID_LOGIN);
+        newUser.setPassword(VALID_PASSWORD);
+        newUser.setAge(VALID_AGE);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(newUser);
+        });
+        assertEquals("User with that login has already exists.Try new login!",
+                exception.getMessage());
+    }
+
+    @Test
+    void register_successRegistration_Ok() {
+        User registeredUser = new User();
+        registeredUser.setLogin(VALID_LOGIN);
+        registeredUser.setPassword(VALID_PASSWORD);
+        registeredUser.setAge(VALID_AGE);
+        User actualUser = registrationService.register(registeredUser);
+        assertEquals(registeredUser, actualUser);
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User userNullLogin = new User();
+        userNullLogin.setLogin(null);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userNullLogin);
+        });
+        assertEquals("Login cannot be null!", exception.getMessage());
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userShortLogin = new User();
+        userShortLogin.setLogin(INVALID_BORDERLINE_LOGIN);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userShortLogin);
+        });
+        assertEquals("Login cannot be less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userShortPassword = new User();
+        userShortPassword.setLogin(VALID_LOGIN);
+        userShortPassword.setPassword(INVALID_BORDERLINE_PASSWORD);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userShortPassword);
+        });
+        assertEquals("Password cannot be less than 6 characters!", exception.getMessage());
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User userNullPassword = new User();
+        userNullPassword.setLogin(VALID_LOGIN);
+        userNullPassword.setPassword(null);
+        ValidationException exception = assertThrows(ValidationException.class, () -> {
+            registrationService.register(userNullPassword);
+        });
+        assertEquals("Password cannot be null!", exception.getMessage());
+    }
+
+    @Test
+    void register_lowAge_notOk() {
+        User userLowAge = new User();
+        userLowAge.setLogin(VALID_LOGIN);
+        userLowAge.setPassword(VALID_PASSWORD);
+        userLowAge.setAge(INVALID_BORDERLINE_AGE);
+        userLowAge.setAge(INVALID_NEGATIVE_AGE);
```

Comment:
why do you reset age?

Reply:
@MentorMateAcademy in the file Common mistakes (jv-registration-validation) it was indicated that I can have several not-ok checks in one not-ok test method. Should I remove them?

Reply:
Correct, but in this case you need to create several users with different not ok ages 
Now in your case you just set up age and then reset different age